doc/source/remote_reading.rst

```diff
@@ -0,0 +1,166 @@
+
+    from satpy import Scene
+
+    storage_options = {'anon': True}
+    filenames = ['s3://noaa-goes16/ABI-L1b-RadC/2019/001/17/*_G16_s20190011702186*']
+    scn = Scene(reader='abi_l1b', filenames=filenames, reader_kwargs={'storage_options': storage_options})
+    scn.load(['true_color_raw'])
+
+Reading from S3 as above requires the `s3fs` library to be installed in addition to `fsspec`.
+
+As an alternative, the storage options can be given using
+`fsspec configuration <https://filesystem-spec.readthedocs.io/en/latest/features.html#configuration>`_.
+For the above example, the configuration could be saved to `s3.json` in the `fsspec` configuration directory
+(by default placed in `~/.config/fsspec/` directory in Linux)::
+
+    {
+        "s3": {
+            "anon": "true"
+        }
+    }
+
+.. note::
+
+    Using `reader_kwargs` doesn't completely override the configuration file, so in case of problems
+    remove the configuration file to see if that solves the issue.
+
+For reference, reading SEVIRI HRIT data from a local S3 storage works the same way::
+
+    filenames = [
+        's3://satellite-data-eumetcast-seviri-rss/H-000-MSG3*202204260855*',
+    ]
+    storage_options = {
+        "client_kwargs": {"endpoint_url": "https://PLACE-YOUR-SERVER-URL-HERE"},
+        "secret": "VERYBIGSECRET",
+        "key": "ACCESSKEY"
+    }
+    scn = Scene(reader='seviri_l1b_hrit', filenames=filenames, reader_kwargs={'storage_options': storage_options})
+    scn.load(['WV_073'])
+
+Using the `fsspec` configuration in `s3.json` the configuration would look like this::
+
+    {
+        "s3": {
+            "client_kwargs": {"endpoint_url": "https://PLACE-YOUR-SERVER-URL-HERE"},
+            "secret": "VERYBIGSECRET",
+            "key": "ACCESSKEY"
+        }
+    }
+
+
+Using multiple readers
+======================
+
+If multiple readers are used and the required credentials differ, the storage options are passed per reader like this::
+
+    reader1_filenames = [...]
+    reader2_filenames = [...]
+    filenames = {
+        'reader1': reader1_filenames,
+        'reader2': reader2_filenames,
+    }
+    reader1_storage_options = {...}
+    reader2_storage_options = {...}
+    reader_kwargs = {
+        'reader1': {
+            'option1': 'foo',
+            'storage_options': reader1_storage_options,
+        },
+        'reader2': {
+            'option1': 'foo',
+            'storage_options': reader1_storage_options,
+        }
+    }
+    scn = Scene(filenames=filenames, reader_kwargs=reader_kwargs)
+
+
+Caching the remote files
+========================
+
+Caching the remote file locally can speedup the overall processing time significantly, especially if the data are re-used
+for example when testing. The caching can be done by taking advantage of the `fsspec caching mechanism
+<https://filesystem-spec.readthedocs.io/en/latest/features.html#caching-files-locally>`_::
```

Comment:
The fsspec docs say:

>  Only “simplecache” is guaranteed thread/process-safe.

Should we maybe only mention `simplecache` here in that case?

Reply:
I tested `filecache` and `blockcache` with the default threaded scheduler using 4 Dask workers and it worked great. I think that's because the cache is filled with full files when the `Scene` is created and that's a single threaded operation. Would the distributed scheduler change that somehow?

Reply:
I could see that depending on the reader. The fact that it says it explicitly and we're talking about concurrency makes me worried that it may be hard to produce an error.

Reply:
Roger, I'll switch to `simplecache`. The cache expiry on `filecache` isn't really something that is a huge consern on satellite data processing.

Reply:
Done.